codegen/codegen.go

```diff
@@ -10,23 +10,39 @@ import (
 	"strings"
 )
 
-func Generate(projectName string) {
+func Generate(projectName, framework string) {
 	fmt.Println("Generating", projectName)
-	for path, srcText := range static.Sources() {
-		srcText = strings.Replace(srcText, "project-name", projectName, -1)
-		binary := []byte(srcText)
-		generateFile(projectName, path, binary)
-	}
-	for path, binary := range static.Images() {
-		generateFile(projectName, path, binary)
-	}
+	processTextFiles(projectName, framework)
+	processImages(projectName, framework)
 	err := initGitRepo(projectName)
 	if err != nil {
 		fmt.Println("Failed to setup a Git repository:", err)
 	}
 	fmt.Println("Generation completed.")
 }
 
+func processImages(projectName string, framework string) {
+	for path, binary := range static.Images() {
+		if !include(path, framework) {
+			continue
+		}
+		path = strings.Replace(path, framework+".", "", 1)
+		generateFile(projectName, path, binary)
+	}
+}
+
+func processTextFiles(projectName string, framework string) {
+	for path, srcText := range static.Sources() {
+		if !include(path, framework) {
+			continue
+		}
+		path = strings.Replace(path, framework+".", "", 1)
```

Comment:
I would suggest to move this line to the `generateFile` Method as it is needed for images as well as text files, this would reduce duplication.

Reply:
Turned out, if I used it in `generateFile` I would have to pass 4 arguments to the function. Instead, I introduced a `Generator` struct that holds project parameters so that I could get `projectName` and `frontend` in any function of `Generator`.